src/lib/Storage/ContentDataSource.php

```diff
@@ -22,59 +23,69 @@
 use Ibexa\Personalization\Value\Storage\Item;
 use Ibexa\Personalization\Value\Storage\ItemList;
 use Ibexa\Personalization\Value\Storage\ItemType;
+use Psr\Log\LoggerInterface;
 
 final class ContentDataSource implements DataSourceInterface
 {
     private SearchService $searchService;
-
     private ContentService $contentService;
-
     private QueryType $queryType;
-
     private DataResolverInterface $dataResolver;
+    private LoggerInterface $logger;
 
     public function __construct(
         SearchService $searchService,
         ContentService $contentService,
         QueryType $queryType,
-        DataResolverInterface $dataResolver
+        DataResolverInterface $dataResolver,
+        LoggerInterface $logger
     ) {
         $this->searchService = $searchService;
         $this->contentService = $contentService;
         $this->queryType = $queryType;
         $this->dataResolver = $dataResolver;
+        $this->logger = $logger;
     }
 
     /**
      * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
      */
     public function countItems(CriteriaInterface $criteria): int
     {
-        $query = $this->queryType->getQuery(['criteria' => $criteria]);
-        $query->limit = 0;
-        $languageFilter = ['languages' => $criteria->getLanguages()];
+        try {
+            $query = $this->queryType->getQuery(['criteria' => $criteria]);
+            $query->limit = 0;
+            $languageFilter = ['languages' => $criteria->getLanguages()];
 
-        return $this->searchService->findContent($query, $languageFilter)->totalCount ?? 0;
+            return $this->searchService->findContent($query, $languageFilter)->totalCount ?? 0;
+        } catch (NotFoundException $exception) {
```

Comment:
In what situation `findContent` throws `NotFoundException`? Repository should not behave like that. It's not even API-reported error.

Reply:
In situation when try to find content but it doesn't exist in given language then below exception is thrown:


```
In Handler.php line 122:
                                                             
  [eZ\Publish\Core\Base\Exceptions\NotFoundException (404)]  
  Could not find 'Language' with identifier 'ger-DE'   
```

Reply:
> In situation when try to find content but it doesn't exist in given language then below exception is thrown:
> 
> ```
> (/vendor/ezsystems/ezplatform-kernel/eZ/Publish/Core/Persistence/Legacy/Content/Language/Handler.php:122)
> In Handler.php line 122:
>                                                              
>   [eZ\Publish\Core\Base\Exceptions\NotFoundException (404)]  
>   Could not find 'Language' with identifier 'ger-DE'   
> ```

Hmm, ok. Looks like Repository Search bug, so let's keep this handling for now.